plugins/repoquery.py

```diff
@@ -222,6 +230,39 @@ def configure(self, args):
         demands.sack_activation = True
         demands.available_repos = True
 
+    def by_all_deps(self, name, query):
+        defaultquery = query.filter(name=name)
+        requiresquery = self.by_requires(self.base.sack, name, query)
+        allpkgs = []
+        for pkg in defaultquery:
+            for reqpkg in requiresquery.run():
+                allpkgs.append(reqpkg)
+
+            # for filePath in pkg.files:
+            #     fileQuery = query.filter(file=filePath)
+            #     print(list(fileQuery))
+            #     for reqpkg in fileQuery.run():
+            #         print(reqpkg.name)
+
+            for provide in pkg.provides:
+                provideQuery = query.filter(requires=provide)
+                for needsProvidePkg in provideQuery.run():
+                    allpkgs.append(needsProvidePkg)
+        allpkgs = self.remove_pkg_duplicates(allpkgs)
+        alldepsquery = query.filter(pkg=allpkgs)
+        return alldepsquery
+
+    def remove_pkg_duplicates(self, array):
```

Comment:
Hm, or even better, replace `allpkgs = []` with `allpkgs = set()` and `allpkgs.append(needsProvidePkg)` with `allpkgs.add(needsProvidePkg)`. You don't need to call `remove_pkg_duplicates` then.